### ISSUE TITLE
Deprecate `_sdl2.window` module

### ISSUE THREAD

**Issue by yunline** [pygame-community/pygame-ce#2401]
`_sdl2.window` is added in #2114 for backward compatibility.
Since the old Window cython sources have been removed, I think it's time to deprecate `_sdl2.window` now.

- [ ] Raise a warning when `_sdl2.window` is imported (2.3.2)
- [ ] Remove `_sdl2.window` (2.4.0)

**Comment by Starbuck5** [comment]
> I think it's time to deprecate _sdl2.window now.

I disagree. The replacement is not "done"-- it's not documented or publicly exposed or finalized. Deprecation of _sdl2(.video).window can happen after pygame.Window or pygame.display.Window is publicly released.

Even once a replacement is fully released, I think we should keep the alias for a long time, as it doesn't hurt us to do so, and helps immensely with backwards compatibility.
